gnpy/core/request.py

```diff
@@ -330,13 +330,13 @@ def compute_constrained_path(network, req):
             # last node which is the transceiver)
             # if all nodes i n node_list are LOOSE constraint, skip the constraints and find
             # a path w/o constraints, else there is no possible path
-            if nodes_list[:-len("STRICT")]:
+            if nodes_list[:-len(['STRICT'])]:
```

Comment:
I'm a bit confused by this code. Firstly, since `len(['STRICT'])` will always be 1, why not just write 1? Secondly, this just checks if there are more than one element in the list, right? But you have already checked if length of `nodes_list` is exactly 1 above and this code is in the else clause so it will always be true. Am I missing something?

Reply:
@ojnas thanks for the feedback ! Indeed the bug comes from an initial error I made using -6 instead of -1, and len('STRICT') instead of 6. nodes_list is the ordered list of include nodes specified by the user + the destination node.
So this first test checks if there are include nodes in the list.  At this stage of the program, there is no path found, and the purpose of this test is to check if there was some constraint added by the user.

- if nodes_list[:-1] not empty -> test is true: the message informs the user about the impossibility to find a path with the list of include nodes
- If nodes_list[:-1] is empty the test is False, and the message should inform the user that there is no path found: so you are right the second message is misleading.

then the second set of tests verifies if the include node list is strict or not (again removing the last element). If no "STRICT" is found in the list then  the constraints are not an obligation, and another path is searched without them.

Is this clearer ?
I can try to improve the test and messages  or do you have any better suggestion ?

Reply:
I have corrected according to your suggestions. is it better ?